upgrading/topics/keycloak/changes.adoc

```diff
@@ -1,5 +1,20 @@
 == Migration Changes
 
+=== Migrating to 3.4.2
+
+==== Added session_state parameter to OpenID Connect Authentication Response
+
+OpenID Connect Session Management specification requires that parameter `session_state` is present in the OpenID Connect Authentication Response.
+In the older versions, we didn't add this parameter, but now {project_name} adds this parameter by default as required per the specification.
+
+Problem is, that some OpenID Connect / OAuth2 adapters and especially older {project_name} adapters may have issues with this new parameter.
+For example parameter will be always present in the browser URL after successful authentication to the client application. For those cases,
+it may be useful to disable adding session_state parameter to the authentication response. It can be done for the particular client in
+{project_name} admin console in client details and section with `OpenID Connect Compatibility Modes` described in <<_compatibility_with_older_adapters>> .
+There is the switch `Exclude Session State From Authentication Response`, which can be turned on to not add the `session_state` parameter
+to the Authentication Response.
+
```

Comment:
The OpenID Connect Session Management specification requires that the parameter `session_state` is present in the OpenID Connect Authentication Response.

In past releases, we did not have this parameter, but now {project_name} adds this parameter by default, as required by the specification.

However, some OpenID Connect / OAuth2 adapters, and especially older {project_name} adapters, may have issues with this new parameter.

For example, the parameter will be always present in the browser URL after successful authentication to the client application. In these cases, it may be useful to disable adding the `session_state` parameter to the authentication response. This can be done for the particular client in the {project_name} admin console, in client details in the section with `OpenID Connect Compatibility Modes`, described in <<_compatibility_with_older_adapters>>. There is the `Exclude Session State From Authentication Response` switch, which can be turned on to prevent adding the `session_state` parameter to the Authentication Response.